incubator/hnc/internal/validators/hierarchy.go

```diff
@@ -210,9 +212,55 @@ func (v *Hierarchy) checkParent(ns, curParent, newParent *forest.Namespace) admi
 		return deny(metav1.StatusReasonConflict, "Illegal parent: "+reason)
 	}
 
+	// Prevent overwriting source objects in the descendants after the hierarchy change.
+	if yes, co := v.hasConflictObject(newParent, ns); yes {
```

Comment:
You _could_ just return `co`:

`if co := v.getOverwrittenObjects(newParent, ns); co != nil {`